exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/ParquetGroupScan.java

```diff
@@ -791,6 +799,43 @@ public FileGroupScan clone(FileSelection selection) throws IOException {
   }
 
   @Override
+  public GroupScan applyLimit(long maxRecords) {
```

Comment:
Suppose the query has LIMIT 1000 and the first set of row groups considered by this rule have a small row count, then in the worst case there could be 1000 files, whereas a single larger file may be sufficient to meet the limit.  Is that something we should consider here ?